server/plugin/command_test.go

```diff
@@ -63,31 +73,178 @@ func TestExecuteCommand(t *testing.T) {
+		},
+		{
+			description:     "ExecuteCommand: boards delete subscription command",
+			isConnected:     true,
+			commandArgs:     &model.CommandArgs{Command: "/azuredevops boards subscription delete mockSubscriptionID"},
+			isDeleteCommand: true,
+			serviceType:     "boards",
+		},
+		{
+			description:                  "ExecuteCommand: failed to delete subscription from store",
+			isConnected:                  true,
+			commandArgs:                  &model.CommandArgs{Command: "/azuredevops boards subscription delete mockSubscriptionID"},
+			isDeleteCommand:              true,
+			serviceType:                  "boards",
+			deleteSubscriptionStoreError: errors.New("failed to delete subscription from store"),
+		},
+		{
+			description:             "ExecuteCommand: failed to get all subscriptions while deleting a subscription",
+			isConnected:             true,
+			commandArgs:             &model.CommandArgs{Command: "/azuredevops boards subscription delete mockSubscriptionID"},
+			isListCommand:           true,
+			serviceType:             "boards",
+			getAllSubscriptionError: errors.New("failed to get all subscriptions while deleting a subscription"),
+			ephemeralMessage:        constants.GenericErrorMessage,
+		},
+		{
+			description:      "ExecuteCommand: boards list subscription command",
+			isConnected:      true,
+			commandArgs:      &model.CommandArgs{Command: "/azuredevops boards subscription list me"},
+			isListCommand:    true,
+			ephemeralMessage: "mockSubscriptionList",
+		},
+		{
+			description:             "ExecuteCommand: failed to get all the subscriptions",
+			isConnected:             true,
+			commandArgs:             &model.CommandArgs{Command: "/azuredevops boards subscription list me"},
+			isListCommand:           true,
+			getAllSubscriptionError: errors.New("failed to get all subscriptions"),
+			ephemeralMessage:        constants.GenericErrorMessage,
+		},
+		{
+			description:      "ExecuteCommand: repos command when user is not connected",
+			commandArgs:      &model.CommandArgs{Command: "/azuredevops repos wrong [title] [description]"},
+			ephemeralMessage: fmt.Sprintf(constants.ConnectAccountFirst, fmt.Sprintf(constants.ConnectAccount, p.GetPluginURLPath(), constants.PathOAuthConnect)),
+		},
+		{
+			description:      "ExecuteCommand: invalid repos command",
+			isConnected:      true,
+			commandArgs:      &model.CommandArgs{Command: "/azuredevops repos wrong [title] [description]"},
+			ephemeralMessage: constants.InvalidCommand + constants.HelpText,
+		},
+		{
+			description: "ExecuteCommand: repos add subscription command",
+			isConnected: true,
+			commandArgs: &model.CommandArgs{Command: "/azuredevops repos subscription add"},
+		},
+		{
+			description:     "ExecuteCommand: repos delete subscription command",
+			isConnected:     true,
+			commandArgs:     &model.CommandArgs{Command: "/azuredevops repos subscription delete mockSubscriptionID"},
+			isDeleteCommand: true,
```

Comment:
i think there is no need for these kind of variables here you can get them from the arguments of the command and same for service type. See if this is possible

Reply:
this will be a little hard to implement. We can discuss this over a call

Reply:
discussed